compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -485,6 +485,16 @@ public class #{typeNameWithParams declName declParams}#{maybe interface baseClas
         #{makeStructBondTypeMember_initialize java declParams structFields structBase}
 
         @Override
+        public final String getName() {
+            return "#{declName}";
+        }
+
+        @Override
+        public final String getQualifiedName() {
+            return "#{getDeclTypeName idl s}";
```

Comment:
This seems off. We have a `qualifiedDeclaredTypeName` in `Java/Util.hs`, but even if you don't want to use it, there's already a `MappingContext` passed to `class_java` called `java`.